core/lsp/helpers.cc

```diff
@@ -0,0 +1,189 @@
+        /* instantiate generic classes */
+        resultType = core::Types::resultTypeAsSeenFrom(gs, resultType, inWhat.enclosingClass(gs), applied->klass,
+                                                       applied->targs);
+    }
+    if (!resultType) {
+        resultType = core::Types::untypedUntracked();
+    }
+    if (receiver) {
+        resultType = core::Types::replaceSelfType(gs, resultType, receiver); // instantiate self types
+    }
+    if (constr) {
+        resultType = core::Types::instantiate(gs, resultType, *constr); // instantiate generic methods
+    }
+    return resultType;
+}
+
+// iff a sig has more than this many parameters, then print it as a multi-line sig.
+constexpr int MAX_PRETTY_SIG_ARGS = 4;
+// iff a `def` would be this wide or wider, expand it to be a multi-line def.
+constexpr int MAX_PRETTY_WIDTH = 80;
+
+string prettySigForMethod(const core::GlobalState &gs, core::MethodRef method, const core::TypePtr &receiver,
+                          core::TypePtr retType, const core::TypeConstraint *constraint,
+                          const bool concretizeIfAbstract = false) {
+    ENFORCE(method.exists());
+    ENFORCE(method.data(gs)->dealiasMethod(gs) == method);
+    // handle this case anyways so that we don't crash in prod when this method is mis-used
+    if (!method.exists()) {
+        return "";
+    }
+
+    if (!retType) {
+        retType = getResultType(gs, method.data(gs)->resultType, method, receiver, constraint);
+    }
+    string methodReturnType =
+        (retType == core::Types::void_()) ? "void" : absl::StrCat("returns(", retType.show(gs), ")");
+    vector<string> typeAndArgNames;
+
+    vector<string> flags;
+    auto sym = method.data(gs);
+    string sigCall = "sig";
+    if (sym->flags.isFinal) {
+        sigCall = "sig(:final)";
+    }
+    if (sym->flags.isAbstract && concretizeIfAbstract) {
+        flags.emplace_back("override");
+    } else if (sym->flags.isAbstract) {
+        flags.emplace_back("abstract");
+    }
+    if (sym->flags.isOverridable) {
+        flags.emplace_back("overridable");
+    }
+    if (sym->flags.isOverride) {
+        flags.emplace_back("override");
+    }
+    for (auto &argSym : method.data(gs)->arguments) {
+        // Don't display synthetic arguments (like blk).
+        if (!argSym.isSyntheticBlockArgument()) {
+            typeAndArgNames.emplace_back(absl::StrCat(
+                argSym.argumentName(gs), ": ", getResultType(gs, argSym.type, method, receiver, constraint).show(gs)));
+        }
+    }
+
+    string flagString = "";
+    if (!flags.empty()) {
+        flagString = fmt::format("{}.", fmt::join(flags, "."));
+    }
+    string paramsString = "";
+    if (!typeAndArgNames.empty()) {
+        paramsString = fmt::format("params({}).", fmt::join(typeAndArgNames, ", "));
+    }
+
+    auto oneline = fmt::format("{} {{{}{}{}}}", sigCall, flagString, paramsString, methodReturnType);
```

Comment:
This block formatting for the `sig` (which looks like `sig {override.void}` for example) goes against the recommendation in most Ruby styleguides ([example](https://github.com/rubocop/ruby-style-guide#spaces-braces)) to add spaces around curly braces that represent an inline block (as opposed to those that represent a Hash, where opinion is more mixed). Might we consider changing this? If so, I will file an issue to track that work.

Reply:
> Might we consider changing this?

Yes, this is a case were the `sig {override.void}` syntax was previously required by Stripe's internal style guide. But about a year ago we changed our style guide to prefer `sig { override.void }` to conform with the rest of the ruby community.

I already changed many of these:

https://github.com/sorbet/sorbet/pull/6572

I guess I missed this one.

Reply:
Thanks for the context. I'll wait until after this lands to file an issue (just so that I can reference code that has actually been integrated). I'll also attempt to address that issue myself as a quick follow-up to this PR.